core/sync.js

```diff
@@ -514,8 +514,9 @@ class Sync {
         await this.pouch.put({
           ...unsynced,
           sides: {
-            [side]: metadata.extractRevNumber(doc) + 1,
-            [other]: unsynced.sides[other] + 1
+            rev: unsynced.sides.rev + 1, // increase rev because of new merge
+            [side]: doc.sides.rev,
+            [other]: unsynced.sides[other] + 1 // increase side to mark change as applied
```

Comment:
Yes unfortunately. Maybe some functions would help by removing the need to remember and documenting it.

Reply:
Actually, `metadata.markSide()` does this job in most places. We can't use it here because we're modifying both sides with different values but this is not the common update to sides.